src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,271 @@
+HANG_REGEX = r"""([^a-zA-Z0-9])\1{3,}"""
+
+# Regex for replacing HTML_Entities
+HTML_ENTITIES_REGEX = r"""&(#?(x?))([^&;\s]+);"""
+
+HANDLES_REGEX = r"""(?x)
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){20}(?!@))
+                |
+                (?<![A-Za-z0-9_!@#\$%&*])@(([A-Za-z0-9_]){1,19})(?![A-Za-z0-9_]*@)
+                """
+
+
+"""
+    replace_html_entities(input_text::AbstractString,
+                    remove_illegal=true) => (entities_replaced_text::AbstractString)
+
+Removes entities from text by converting them to their corresponding unicode character.
+`input_text::AbstractString` The string on which HTML entities need to be replaced
+`remove_illegal::Bool` If `True`, entities that can't be converted are
+removed. Otherwise, entities that can't be converted are kept "as
+is".
+Returns `entities_replaced_text::AbstractString`
+"""
+function replace_html_entities(input_text::AbstractString, remove_illegal=true)
+
+    function convert_entity(matched_text)
+
+        groups = match(HTML_ENTITIES_REGEX, matched_text).captures
+        entity_body = groups[3]
+        local number::Number = 0
+        if isempty(groups[1])
+            return(lookupname(HTML_Entities.default, entity_body))
+        else
+            if isempty(groups[2])
+                is_numeric = true
+                for i in entity_body
+                    if !isdigit(i)
+                        is_numeric = false
+                        break
+                    end
+                end
+                if is_numeric
+                    number = parse(Int, entity_body, base=10)
+                end
+            else
+                is_base_16 = true
+                allowed_letters = ['a', 'b', 'c', 'd', 'e', 'f']
+                for i in entity_body
+                    if !(isdigit(i) || i in allowed_letters)
+                        is_base_16 = false
+                        break
+                    end
+                end
+                if is_base_16
+                    number = parse(Int, entity_body, base=16)
+                end
+            end
+
+            # Numeric character references in the 80-9F range are typically
+            # interpreted by browsers as representing the characters mapped
+            # to bytes 80-9F in the Windows-1252 encoding. For more info
+            # see: http://en.wikipedia.org/wiki/Character_encodings_in_HTML
+
+            if 0x80 <= number <= 0x9F
+                if !(number in [129 141 143 144 157])
+                    return decode([UInt8(number)], "WINDOWS-1252")
+                end
+            else
+                if Unicode.isassigned(number)
+                    return (Char(number))
+                end
+            end
+        end
+
+        if remove_illegal
+            return ""
+        else
+            return matched_text
+        end
+    end
+
+    entities_replaced_text = replace(input_text, HTML_ENTITIES_REGEX => convert_entity)
```

Comment:
```suggestion
    entities_replaced_text = replace(input_text, HTML_ENTITIES_REGEX => convert_entity)
    return entities_replaced_text
```

Explict return in long function aids readbility